test/software/amazon/ion/impl/lite/BaseIonSequenceLiteTestCase.java

```diff
@@ -0,0 +1,655 @@
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        final List<IonValue> insideSubList = Arrays.asList(parent.get(2), parent.get(3));
+        final List<IonValue> outsideSublist = Arrays.asList(parent.get(0), parent.get(1));
+        final List<IonValue> mixed = Arrays.asList(parent.get(0), parent.get(3));
+
+        assertTrue(sublist.containsAll(insideSubList));
+        assertFalse(sublist.containsAll(outsideSublist));
+        assertFalse(sublist.containsAll(mixed));
+    }
+
+    @Test(expected = ConcurrentModificationException.class)
+    public void sublistContainsConcurrentModification() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+        final IonValue value = parent.remove(0);
+
+        sublist.contains(Collections.singletonList(value));
+    }
+
+    @Test
+    public void sublistToArray() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        final Object[] array = sublist.toArray();
+        assertEquals(3, array.length);
+        assertEquals(2, ((IonInt) array[0]).intValue());
+        assertEquals(3, ((IonInt) array[1]).intValue());
+        assertEquals(4, ((IonInt) array[2]).intValue());
+    }
+
+    @Test(expected = ConcurrentModificationException.class)
+    public void sublistToArrayConcurrentModification() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+        parent.remove(0);
+
+        sublist.toArray();
+    }
+
+    @Test
+    public void sublistToTypedArray() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        final IonValue[] array = sublist.toArray(IonValue.EMPTY_ARRAY);
+
+        assertEquals(3, array.length);
+        assertEquals(2, ((IonInt) array[0]).intValue());
+        assertEquals(3, ((IonInt) array[1]).intValue());
+        assertEquals(4, ((IonInt) array[2]).intValue());
+    }
+
+    @Test(expected = ConcurrentModificationException.class)
+    public void sublistToTypedArrayConcurrentModification() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+        parent.remove(0);
+
+        sublist.toArray(IonValue.EMPTY_ARRAY);
+    }
+
+    @Test
+    public void sublistAdd() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        final IonInt value = SYSTEM.newInt(99);
+        sublist.add(value);
+        assertEquals(4, sublist.size());
+        assertEquals(value, sublist.get(3));
+    }
+
+    @Test
+    public void sublistAddSame() {
```

Comment:
this test appears to be testing the same thing as `sublistAdd()` -- what value is it adding?

Reply:
Just noticed this test is wrong since `IonSequence` uses reference equality. `Collections#add` contract is to return `false` when trying to add the same value to a collection that does not support having duplicates. 

`IonSequenceLite` actually throws when trying to add a duplicate, will change the test to reflect that.